android/lib/src/main/java/org/vosk/android/SpeechService.java

```diff
@@ -227,31 +253,91 @@ public void run() {
                 if (nread < 0)
                     throw new RuntimeException("error reading audio buffer");
 
-                if (recognizer.acceptWaveForm(buffer, nread)) {
-                    final String result = recognizer.getResult();
-                    mainHandler.post(() -> listener.onResult(result));
-                } else {
-                    final String partialResult = recognizer.getPartialResult();
-                    mainHandler.post(() -> listener.onPartialResult(partialResult));
+                // Adds the data read from the AudioRecorder to the queue of chunks of audio
+                try {
+                    transcriptionThread.recordingChunksQueue.put(new RecordingChunk(buffer, nread));
+                } catch (InterruptedException e) {
+                    interrupt();
                 }
 
-                if (timeoutSamples != NO_TIMEOUT) {
-                    remainingSamples = remainingSamples - nread;
-                }
             }
-
             recorder.stop();
+        }
+
+        @Override
+        public void interrupt() {
+            transcriptionThread.interrupt();
+            super.interrupt();
+        }
+    }
+
+    /**
+     * Class used to save the audio buffer got from AudioRecorder and the respective number of
+     * shorts read
+     * @param audioBuffer - contains the buffer of audio
+     * @param nread - number of shorts read
+     *
+     */
+    private static final class RecordingChunk {
+        private final int nread;
+        private final short[] audioBuffer;
 
-            if (!paused) {
-                // If we met timeout signal that speech ended
-                if (timeoutSamples != NO_TIMEOUT && remainingSamples <= 0) {
-                    mainHandler.post(() -> listener.onTimeout());
-                } else {
-                    final String finalResult = recognizer.getFinalResult();
-                    mainHandler.post(() -> listener.onFinalResult(finalResult));
```

Comment:
we no longer have the onTImeout() and onFinalResult() callbacks, are they not necessary?

Reply:
Well we don't use these callbacks and therefore I'm removing them in order to not get possible overhead running them without being necessary.